### PR TITLE
added package.json to priv specifying type: "commonjs"

### PR DESCRIPTION
**Pull request by Valian**
In my Elixir project, I have package.json in a root directory. It specifies `type: module` as I'm writing JS in ESM syntax, with import / export. With that setup, nodejs treats `deps/nodejs/priv/server.js` as a project script, since it doesn't have package.json next to it. Exact error I'm getting:

> ReferenceError: require is not defined in ES module scope, you can use import instead
This file is being treated as an ES module because it has a '.js' file extension and '/Users/user/project/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.

Adding package.json to priv simply explicitly informs nodejs that script is written in CJS.

### CODE REVIEW

**Comment by Valian** [comment]
@grossvogel this is a very small change ;) without that package.json LiveVue projects treats scripts in exixir-nodejs as ESM and errors out. I'm using that branch for a long time already and it seems be working just fine.

**Review by grossvogel — APPROVED** [review]

**Comment by grossvogel** [comment]
Thanks, @Valian, I'd like one other person to give this a quick look, then we'll get it merged!

**Comment by Valian** [comment]
Sure! Looking forward to it :+1: Thanks for finding time, I know it's tough 😅

File: priv/package.json [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
+  "name": "elixir-nodejs",
+  "type": "commonjs"
+}
```
Comment (Matt-Cain) [inline priv/package.json:4]: I think it's looking for a new line at the end here.

Comment (Valian) [reply priv/package.json:4]: Yes, right. That's how Github shows lack of newline at the end of the file. I'll add it

Comment (Valian) [reply priv/package.json]: Done

**Review by Matt-Cain — APPROVED** [review]
I'm not super knowledgable in this area. Was wondering if you had tried the suggestion in the error: `"To treat it as a CommonJS script, rename it to use the '.cjs' file extension."` as that may be better than adding another file 🤷‍♂️. 

I also don't see any issue with this approach so I'm good with it 👍

**Comment by Valian** [comment]
Thank you for approval! Looking forward for a patch release :) 

To my knowledge, NodeJS is rather dumb when it comes to deciding if `.js` extension should be loaded as ESM (imports / exports) or CJS (require). A short summary given by chatgpt can be found [here](https://chatgpt.com/share/674e38d4-6d58-800a-ac80-645a397c560e). 

In my particular example node thought server.js is ESM because root of my project has `package.json` with `type: module`, and it's the first found package.json.

**Comment by andresgutgon** [comment]
>In my particular example node thought server.js is ESM because root of my project has package.json with type: module, and it's the first found package.json.

Hi @Valian is your project a phoenix project? I was wondering if that's true how so you have the package.json at the root of your Elixir project? how you organize your JS code?

**Comment by Valian** [comment]
@andresgutgon  Yes, project I'm working on is a Phoenix project using my package https://github.com/Valian/live_vue. 

By default, phoenix project has all the JS assets in the assets directory and no package.json. Usually when something is needed from NPM people install it also inside assets, but it's just a convention. 

In my case, I'm colocating Vue components with LiveViews, as my views always have a single top-level Vue component. It's the same approach as Inertia.js, but with Phoenix LiveView backing props synchronization and event handling. 

Just, if node_modules are inside `assets`, then these colocated components can't correctly import npm packages. So I've moved it to the root of the project and stumbled upon that problem. Other than that, everything just works fine :+1:

PS. I'm working on docs update and igniter installer in `live_vue` to also describe that use case. Would you be interested about hearing more?

**Comment by andresgutgon** [comment]
> PS. I'm working on docs update and igniter installer in live_vue to also describe that use case. Would you be interested in hearing more?

I really appreciate your work in this repo it's being super helpful. In my case I'm trying Inertia with Vite and trying to make it work with ESM modules. 

So with this comment you solved my other issue. When I put in `priv/ssr.js` the inertia SSR node code it doesn't have access to `node_modules`. I might just do what you did and move all my JS stuff to the root. 

The only thing I don't like about this is polluting Phoenix root with all the nodejs mambo jambo. Here you can see what the options are I'm contemplating:

https://github.com/inertiajs/inertia-phoenix/issues/37#issuecomment-2726712663

**Comment by Valian** [comment]
@andresgutgon I've commented in the other repo. In short, I've spent way too much time figuring out how to make Vite work seamlessly with Phoenix, including HMR and SSR 😂 Hopefully I'll be able to help you a bit.

**Pull request merged**